lib/Share/Helper/ShareAPIController.php

```diff
@@ -91,25 +91,32 @@ public function formatShare(IShare $share): array {
 			return $result;
 		}
 
-		// The display name of one-to-one rooms is set to the display name of
-		// the other participant, except on reshares to rooms that the user is
-		// not invited to, in which case the display name of both participants
-		// is used.
 		$roomName = $room->getName();
-		if ($room->getType() === Room::ONE_TO_ONE_CALL) {
-			$userIds = $room->getParticipantUserIds();
-			foreach ($userIds as $userId) {
-				if ($this->userId !== $userId) {
+		try {
+			$room->getParticipant($this->userId);
+
+			if ($room->getType() === Room::ONE_TO_ONE_CALL) {
+				$userIds = $room->getParticipantUserIds();
+				foreach ($userIds as $userId) {
+					if ($this->userId === $userId) {
+						continue;
+					}
+
 					$user = $this->userManager->get($userId);
 					if ($user instanceof IUser) {
-						$roomName = $roomName? $roomName . $this->l->t(', ') . $user->getDisplayName(): $user->getDisplayName();
+						$roomName = $user->getDisplayName();
+						break;
 					}
 				}
+			} else if ($roomName === '') {
+				$roomName = $this->l->t('Unnamed conversation');
```

Comment:
When the room has no name I would either return the list of participants (which would be consistent with how unnamed rooms are shown in the main Talk UI) or an empty string so clients can decide the best way to show the lack of name.

For example, in the share tab of the Files app, when a file is shared with a user in a room with no name now it shows _Shared with you and the conversation Unnamed conversation by user0_; before it showed _Shared with you in a conversation by user0_, which in my opinion is better.

Reply:
Yeah, but before that, the list of (re-)shares listed it as ` (conversation)` with an empty avatar. Certainly a worse option.
![bildschirmfoto von 2019-01-31 11-58-15](https://user-images.githubusercontent.com/213943/52050023-90977580-254f-11e9-83e8-4dd9541cccd9.png)

Anyway, I don't think neither is too important and in the future we should prevent empty room names, as they just cause too much weird behaviour

Reply:
>  Yeah, but before that, the list of (re-)shares listed it as (conversation) with an empty avatar. Certainly a worse option.

I agree, but that was a bug in the Sharing tab, not in Talk itself.

>  Anyway, I don't think neither is too important and in the future we should prevent empty room names, as they just cause too much weird behaviour

OK.